std/java/_std/StringBuf.hx

```diff
@@ -52,7 +52,12 @@ class StringBuf {
 	}
 
 	public function addChar( c : Int ) : Void untyped {
-		b.append(cast(c, java.StdTypes.Char16));
+		if (c >= 0x10000) {
```

Comment:
Why not [`appendCodepoint`](https://docs.oracle.com/javase/7/docs/api/java/lang/StringBuilder.html#appendCodePoint(int))?

Reply:
Good point, didn't know about it. I don't think there's anything like it in C# though.

Reply:
Didn't find anything myself either ;)